File: swagger/apis/enrollment_api.py
from .base_api import BaseApi

from ..models.add_client_to_enrollment_request import AddClientToEnrollmentRequest
from ..models.class_schedule import ClassSchedule
from ..models.get_enrollments_response import GetEnrollmentsResponse

class EnrollmentApi(BaseApi):
	def __init__(self, site_id, authorization):
		super().__init__('enrollment', site_id, authorization)

	def post_addclienttoenrollment(self, request):
		pass

	def get_enrollments(self, classScheduleIds, endDate, limit, locationIds, offset, programIds, sessionTypeIds, staffIds, startDate):
		pass

File: swagger/apis/EnrollmentApi.php
<?php

namespace App\Mindbody\Apis;

use App\Mindbody\Models\AddClientToEnrollmentRequest;
use App\Mindbody\Models\ClassSchedule;
use App\Mindbody\Models\GetEnrollmentsRequest;
use App\Mindbody\Models\GetEnrollmentsResponse;

class EnrollmentApi extends BaseApi
{
	/**
	 * Constructor
	 */
	public function __construct($siteId, $authorization)
	{
		parent::__construct('enrollment', $siteId, $authorization);
	}

	/**
	 * Book a client into an enrollment.
	 */
	public function addClientToEnrollment($request)
	{
		$url = $this->getFullPath('addclienttoenrollment');

		return $this->client->post($url, $request, ClassSchedule::class);
	}

	/**
	 * Returns a list of enrollments. An enrollment is a service, such as a
	 * workshop or an event, that a staff member offers to multiple students,
	 * who commit to coming to all or most of the scheduled sessions.
	 * Enrollments typically run for a limited time only.
	 */
	public function getEnrollments($request)
	{
		$url = $this->getFullPath('enrollments');

		return $this->client->get($url, $request, GetEnrollmentsResponse::class);
	}

}
